### ISSUE TITLE
Warn of mutated loop variables

### ISSUE THREAD

**Issue by Zash** [lunarmodules/luacheck#109]
Lua development version has made loop variables read-only in https://github.com/lua/lua/commit/b2f7b3b79f3117885b265575f6c5dbf934757797 so having a warning for this would be good so that those cases could be found early.

E.g. code like

``` lua
for i = 1, 10 do
  i = i - 1
end
for k,v in pairs({ foo = "bar" }) do
  k = "k:"..k
end
```

Fails with `attempt to assign to const variable 'i'`

**Comment by asherber** [comment]
Similarly, a warning when trying to explicitly assign to a const variable in 5.4.

```lua
local foo <const> = 5
foo = 6  -- should generate warning
```

**Comment by alerque** [comment]
Contributions welcome. I don't have a lot of time to develop this myself right now but I mostly keep up with facilitating community contributions.

**Comment by alerque** [comment]
@asherber While the implementation might share some code, the `<const>` issue for Lua 5.4 should probably be tracked in a separate issue. Can you open a feature request for that to be added to the `lua54` and `lua54c` presets?

**Comment by asherber** [comment]
Done: https://github.com/lunarmodules/luacheck/issues/116
